src/linker/Linker/CustomAttributeSource.cs

```diff
@@ -37,7 +37,14 @@ public static AssemblyDefinition GetAssemblyFromCustomAttributeProvider (ICustom
 
 		public bool TryGetEmbeddedXmlInfo (ICustomAttributeProvider provider, [NotNullWhen (true)] out AttributeInfo? xmlInfo)
 		{
-			var assembly = GetAssemblyFromCustomAttributeProvider (provider);
+			AssemblyDefinition? assembly;
+			try {
+				// Might be the internal RemoveAttributesInstancesAttribute which has no Module or Assembly and throws a NullRef
+				assembly = GetAssemblyFromCustomAttributeProvider (provider);
+			} catch (NullReferenceException) {
+				xmlInfo = null;
+				return false;
+			}
```

Comment:
Exceptions should not be used for unexceptional cases. If the code ran into the case where the provider is the made-up attribute, then we should either check for that specifically or make the code able to handle such case.

Reply:
I went to fix this, but I can't reproduce the exception anymore, so I removed the try/catch